File rename

CiprianRegus opened the pull request:
- Moved header files from include to include/no_os directory.
- Updated header and source files with the correct include paths.
- Updated the include paths in src.mk files.

Signed-off-by: Ciprian Regus <ciprian.regus@analog.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/analogdevicesinc/no-OS?pullRequest=1178) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/analogdevicesinc/no-OS?pullRequest=1178) before we can accept your contribution.<br/><hr/>**Ciprian Regus** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/analogdevicesinc/no-OS?pullRequest=1178) it.</sub>

amiclaus reviewed (commented):
nit: some commit descriptions would be nice :)
Also please sign the CLA since you're not yet part of the org.

The CI detected issues seem to be unrelated to the changes, so it should be fine like this.

adrimbarean commented:
nitpick: looks like ad9361_conv.c and ad9361_util.c could be AStyled. I'd do this in a separate commit, but since it's not originally part of this pull request we can also skip it. Choice is yours.

adrimbarean reviewed (approved):
Looks good to me. Just a comment.

mchindri commented on drivers/platform/linux/linux_uart.c:
```diff
@@ -209,7 +209,7 @@ int32_t uart_init(struct uart_desc **desc, struct uart_init_param *param)
 	}
 
-	if (param->stop == UART_STOP_1)
+	if (param->stop == UART_STOP_1BIT)
```
Add in the commit description the reason of the change too

mchindri replied:
Same for previous commit

CiprianRegus commented:
V3:
- AStyled ad9361_conv.c and ad9361_util.c.

buha reviewed (changes requested):
This breaks the project_build target. A little bit of tweaking is needed for those generic.mk changes to make this work with the builds from SDK.

buha commented:
![Screenshot from 2021-12-15 09-40-41](https://user-images.githubusercontent.com/8933663/146146373-de2dd355-38c8-4c03-aed0-006e5ecf4029.png)
![Screenshot from 2021-12-15 09-42-06](https://user-images.githubusercontent.com/8933663/146146379-e5a30242-11f5-4f38-a18d-aab4d4be0295.png)

dbogdan commented:
I would use the same name as the one of the repository: "no-os" instead of "no_os".

amiclaus reviewed (commented):
@mchindri is it possible to add your commits in a separate PR?

buha commented:
I tested projects running on xilinx and stm32 and the include problem in SDK was fixed by your changes

remaining changes:
- squash the commits by Mihail into your big commit that updates generic.mk
- edit ad5758 project and driver which were merged meanwhile to master and don't contain no-os/....h path

CiprianRegus commented:
V4:
- Edited drivers/dac/ad5758 files to include the correct paths.
- Squashed commits that modify generic.mk.
- Changed the commit title to reflect the changed files.

CiprianRegus commented:
V5:
- Solved conflicts with recent master commits.
- Moved include/no-os/uart.h file to the correct commit.

buha commented on include/no-os/uart.h:
```diff
@@ -0,0 +1,162 @@
+/***************************************************************************//**
```
this file seems to have been deleted in a previous commit, only to readded it here, this is not ok

CiprianRegus commented:
V6:
- Fixed include paths in drivers/platform/stm32.
- Solved the issue with the deleted uart.h file in a previous commit.
